native/common/jp_classhints.cpp

```diff
@@ -21,6 +21,55 @@
 #include "jp_stringtype.h"
 #include "pyjp.h"
 
+JPMatch::JPMatch()
+{
+	conversion = NULL;
+	frame = NULL;
+	object = NULL;
+	type = JPMatch::_none;
+	slot = (JPValue*) - 1;
+	closure = 0;
+}
+
+JPMatch::JPMatch(JPJavaFrame *fr, PyObject *obj)
+{
+	conversion = NULL;
+	frame = fr;
+	object = obj;
+	type = JPMatch::_none;
+	slot = (JPValue*) - 1;
+	closure = 0;
+}
+
+JPValue *JPMatch::getJavaSlot()
+{
+	if (slot == (JPValue*) - 1)
+		return slot = PyJPValue_getJavaSlot(object);
```

Comment:
This is one part that I don't really like.  I need to be able to check if the slot has been checked and I am not willing to make another bool (which will cost a whole word because of alignment) to just indicate if the slot has been checked.   Thus we have a 3 state value (-1 not checked, 0 no java slot, and otherwise) has value.   

In the horribly unlikely scenario that the java slot is located at the unaligned address -1 we will look it up multiple times taking a performance hit.